homeassistant/components/octoprint/config_flow.py

```diff
@@ -0,0 +1,197 @@
+            )
+        except ApiError as ex:
+            _LOGGER.error("Failed to retrieve application key: %s", ex)
+            raise InvalidAuth from ex
+
+    octoprint.set_api_key(data[CONF_API_KEY])
+    if not await validate_connection(octoprint):
+        _LOGGER.error("Failed to connect")
+        raise CannotConnect
+
+    discovery = await octoprint.get_discovery_info()
+    uuid = None
+    if discovery:
+        uuid = discovery.upnp_uuid
+
+    # Return info that you want to store in the config entry.
+    return {"title": data[CONF_NAME], "uuid": uuid}
+
+
+async def validate_connection(octoprint: OctoprintClient):
+    """Validate the connection to the printer."""
+    try:
+        await octoprint.get_server_info()
+    except requests.exceptions.RequestException as conn_err:
+        _LOGGER.error("Error setting up OctoPrint API: %r", conn_err)
+        raise CannotConnect from conn_err
+
+    return True
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for OctoPrint."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Handle a config flow for OctoPrint."""
+        self.discovery_schema = None
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        if user_input is None:
+            data = self.discovery_schema or _schema_with_defaults()
+            return self.async_show_form(step_id="user", data_schema=data)
+
+        errors = {}
+
+        try:
+            info = await validate_input(self.hass, user_input)
+        except CannotConnect:
+            errors["base"] = "cannot_connect"
+        except InvalidAuth:
+            errors["base"] = "invalid_auth"
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception")
+            errors["base"] = "unknown"
+        else:
+            if info["uuid"]:
+                await self.async_set_unique_id(info["uuid"])
+                self._abort_if_unique_id_configured()
+            return self.async_create_entry(title=info["title"], data=user_input)
+
+        return self.async_show_form(
+            step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
+        )
+
+    async def async_step_import(self, user_input):
+        """Handle import."""
+        if not user_input:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_IMPORT_DATA_SCHEMA
+            )
+
+        return await self.async_step_user(user_input)
+
+    async def async_step_discovery(self, discovery_info):
+        """Handle discovery flow."""
+        if "udn" in discovery_info and discovery_info["udn"][0:4] == "uuid":
+            await self.async_set_unique_id(discovery_info["udn"][5:])
+            self._abort_if_unique_id_configured()
+
+        # pylint: disable=no-member # https://github.com/PyCQA/pylint/issues/3167
```

Comment:
This may have been resolved in pylint now?

Reply:
Still giving me the error when I run it

Reply:
Dropped legacy discovery in favor of ssdp and zeroconf

Reply:
Try rebasing on latest dev branch and update all our test deps. We have removed this pylint disable from our existing integrations on dev branch.

Reply:
I don't think I have permissions to rebase the base branch this is against. If someone with permissions can rebase the target branch, rework-octoprint, I can rebase against that to drop these.

Reply:
Right. Forgot about that. I'll do it tomorrow.

Reply:
https://github.com/home-assistant/core/tree/rework-octoprint is rebased on latest dev branch.

Reply:
Rebaseed and pushed